.eslintrc.js

```diff
@@ -25,6 +26,20 @@ module.exports = {
         lib: "never",
       },
     ],
+    "import/no-extraneous-dependencies": [
```

Comment:
This linting rule disallows the use packages that are not included in the **nearest** ancestor package.json file.

This has two purposes:
1. Prevent accidental use of development dependencies in files meant for production. We don't want to accidentally bundle `@faker-js/faker`—it's big!
2. Ensure workspaces declare the modules they use as dependencies.

Regarding (2):

Our frontend code is a monorepo with each module specifying its own dependencies. Node module installers (yarn, npm) [hoist](https://classic.yarnpkg.com/blog/2018/02/15/nohoist/) packages to the root `node_modules/` folder in order to de-duplicate packages. Below, the `root` node_modules is thus guaranteed to exist, but the per-package node_modules may or may not exist after a `yarn install`.

```
mit-open (repo)
package.json                <-- root package.json
node_modules                <-- root node_modules
|--frontends/
    |--mit-open/
        package.json           <-- workspace package.json
        node_modules (MAYBE)   <-- workspace node_modules
    |--ol-util/
        package.json
        node_modules
    |--ol-ckeditor/
        package.json
        node_modules
```

But Node's module resolution algorithm searches all `node_modules/` folders in a files ancestor containing directories. Together with hoisting, this means that dependencies in one workspace are often available in second workspace, even if the second does not specify the dependency.

This ESLint rule prevents that by demanding dependencies appear in the **nearest** ancestor package.json.